src/web/app/components/account-requests-table/admin-reject-with-reason-modal/admin-reject-with-reason-modal.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { RejectWithReasonModalComponentResult } from './admin-reject-with-reason-modal-model';
+import { StatusMessageService } from '../../../../services/status-message.service';
+
+/**
+ * Modal to select reject account requests with reason.
+ */
+@Component({
+  selector: 'tm-reject-with-reason-modal',
+  templateUrl: './admin-reject-with-reason-modal.component.html',
+  styleUrls: ['./admin-reject-with-reason-modal.component.scss'],
+})
+
+export class RejectWithReasonModalComponent implements OnInit {
+
+  @Input()
+  accountRequestName: string = '';
+
+  @Input()
+  accountRequestEmail: string = '';
+
+  rejectionReasonBody: string = '<p>Hi, {accountRequestName} </p>\n\n'
+  + '<p>Thanks for your interest in using TEAMMATES. '
+  + 'We are unable to create a TEAMMATES instructor account for you.</p>'
+  + '<p><strong>Reason:</strong> The email address you provided is not an &#39;official&#39; '
+  + 'email address provided by your institution.<br />'
+  + '<strong>Remedy:</strong> Please re-submit an account request with your &#39;official&#39; '
+  + 'institution email address.</p>\n\n'
+  + '<p><strong>Reason:</strong> The email address you have provided does seems like it belongs to a student '
+  + '(i.e., not a staff member) of your institution.<br />'
+  + '<strong>Remedy:</strong> If you are a student but you still need an instructor account, '
+  + 'please send your justification to teammates&#64;comp.nus.edu.sg.</p>\n\n'
+  + '<p><strong>Reason:</strong> You already have an account for this email address and this institution.<br />'
+  + '<strong>Remedy:</strong> You can login to TEAMMATES using your Google account {existingEmail} </p>\n\n'
+  + '<p>If you need further clarification or would like to appeal this decision, please '
+  + 'feel free to contact us at teammates&#64;comp.nus.edu.sg.</p>'
+  + '<p>Regards,<br />TEAMMATES Team.</p>';
+  rejectionReasonTitle: string = 'We are Unable to Create an Account for you';
+
+  constructor(public activeModal: NgbActiveModal, public statusMessageService: StatusMessageService) {}
+
+  ngOnInit(): void {
+    this.rejectionReasonBody = this.rejectionReasonBody.replace('{accountRequestName}', this.accountRequestName);
+    this.rejectionReasonBody = this.rejectionReasonBody.replace('{existingEmail}', this.accountRequestEmail);
+  }
+
+  onRejectionReasonBodyChange(updatedText: string): void {
+    this.rejectionReasonBody = updatedText;
+  }
+
+  /**
+   * Fires the reject event.
+   */
+  reject(): void {
+
+    if (!this.rejectionReasonBody) {
+      this.statusMessageService.showErrorToast('Please provide a reason for rejection.');
```

Comment:
Just a suggestion for this error message: since this error is thrown not only when the reason is empty but because the entire email body is empty.
```suggestion
      this.statusMessageService.showErrorToast('Please provide an email body for the rejection email.');
```